server/api/src/rsDataObjRepl.cpp

```diff
@@ -690,9 +692,11 @@ namespace
         auto source_cond_input = irods::experimental::make_key_value_proxy(source_inp.condInput);
         auto source_obj = resolve_hierarchy_and_get_data_object_info(_comm, source_inp, irods::OPEN_OPERATION);
 
+        const auto* source_hierarchy = source_cond_input.at(RESC_HIER_STR_KW).value().data();
```

Comment:
Is the type of `.value()` a std::string or std::string_view?

If it's a std::string, prefer `.c_str()` instead of `.data()`.

Reply:
`value()` is a `std::string_view`

Reply:
That should be okay as long as we aren't doing any string manipulation on the string_view and the string is null-terminated.

The null-terminated requirement is covered due to the implementation. I think we're good on the string manipulation too. I don't remember seeing the strings change in any way.

You agree?

Reply:
Correct, we do not change the hierarchy string once it has been resolved in this function, at least not in the input structure. The string comes from the hierarchy resolution which I believe is using `std::string` (and/or `irods::hierarchy_parser`, which uses `std::string`), so the string should be null terminated.

We can always make a copy if that seems safer.

Reply:
If the string_view is not being modified (e.g. .substr is called), then we're good.

Feel free to mark as resolved.